### PR TITLE
Password optional widget sign in guide (dotnet)

### PR DESCRIPTION
**Pull request by danielmaharry-okta**
## Description:
- **What's changed?** First draft of guide to password-optional sign-in flow using the sign-in widget  for dotnet
- **Is this PR related to a Monolith release?** No

### Resolves:

* [OKTA-502088](https://oktainc.atlassian.net/browse/OKTA-502088)

NETLIFY Link (Sep 12) : https://631f403700132e1a6b8be9e8--reverent-murdock-829d24.netlify.app/docs/guides/pwd-optional-widget-sign-in-email/aspnet/main/

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+```
+
+> **Note**: For more information on magic links and OTP, including customizations and complete user journeys, see [Email Magic Links Overview](/docs/guides/email-magic-links-overview/main/).
+
+### 4. Your app handles an authentication success response
+
+After the user successfully verifies their identity, Identity Engine sends an interaction code in a query parameter to `${signInRedirectURI}`. For example, `http://localhost:44314/interactioncode/callback?interaction_code=2JFmObNY8snovJP6_UK5gI_l7RQ-....`
+
+Create an endpoint that calls `idxClient.RedeemInteractionCodeAsync()` to exchange the interaction code for access tokens and `AuthenticationHelper.GetIdentityFromTokenResponseAsync()` to retrieve the OIDC claims information about the user and pass them into your application. The user has now signed in.
```
Comment (chrisdmills-okta) [inline packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md:29]: "retrieve the OIDC claims information about the user" is a little awkward. How about " retrieve the user's OIDC claims information" ?

Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md:29]: "pass them into your application" -> by them, are you talking about the OIDC claims information, or the access tokens? If the former, you should change "them" to "it". If the latter, you should make that clearer.

**Review by chrisdmills-okta — APPROVED** [review]
Hi @danielmaharry-okta ! First one down. Nothing major, just a few grammar and style nits mostly. I've approved so as not to hold you up.

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/nodeexpress/bestpractices.md [hunk 2]
```diff
@@ -0,0 +1 @@
+[Recommended best practices](/docs/guides/pwd-optional-best-practices/nodeexpress/main/)
```
Comment (robertnicolo-okta) [inline packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/nodeexpress/bestpractices.md:1]: There is a gap between the lines. See the ../docs/guides/pwd-optional-sign-in-email/nodeexpress/main/ page in the browser.

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/react/bestpractices.md [hunk 3]
```diff
@@ -0,0 +1 @@
+[Recommended best practices](/docs/guides/pwd-optional-best-practices/react/main/)
```
Comment (robertnicolo-okta) [inline packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/react/bestpractices.md:1]: Another gap here.

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/bestpractices.md [hunk 4]
```diff
@@ -0,0 +1 @@
+[Recommended best practices](/docs/guides/pwd-optional-best-practices/aspnet/main/)
```
Comment (robertnicolo-okta) [inline packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/bestpractices.md:1]: An another here.

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+The following code defines an endpoint that retrieves the current Sign-In Widget configuration details from session state and appends the `OTP` and `state` values from the callback to it. If it cannot find the configuration details in session state, it assumes the callback has not come from the same browser and prompts the user accordingly.
```
Comment (robertnicolo-okta) [inline packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md:1]: Agreeing with Chris with the slight awkwardness of the sentence. A possible alternative is 👍 

"The following code defines an endpoint that receives the OTP and state values from the magic link. The code assigns these OTP and state values to the Sign-in Widget configuration object stored in session."

**Review by robertnicolo-okta — CHANGES_REQUESTED** [review]
Looks good Dan. Had a couple of feedback items.

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/index.md [hunk 6]
```diff
@@ -26,7 +26,7 @@ Enable an email-only sign-in flow in your application using the Embedded SDK.
 
 ## Update configurations
 
-<StackSnippet snippet="setupoktaorg" inline/>
+Before you can start integrating password-optional sign-up flows in your app, <StackSnippet snippet="setupoktaorg" inline/>. See also <StackSnippet snippet="bestpractices" inline />.
```
[the remark this answers is not in the document]
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/index.md:29]: Good spot - fixed

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md:1]: My bad wording. We're adding to the config details pulled from session state. Have reworded to "The endpoint defined in the following code attempts to retrieve the current Sign-In Widget configuration details from session state. If successful, the callback's `OTP` and `state` values are added to the configuration details and passed to the Widget. If unsuccessful, the code assumes the callback has not come from the same browser and prompts the user accordingly."

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+      Message = $"Please enter the OTP '{otp}' in the original browser tab to finish the flow."
+   });
+}
+```
+
+> **Note**: For more information on magic links and OTP, including customizations and complete user journeys, see [Email Magic Links Overview](/docs/guides/email-magic-links-overview/main/).
+
+### 4. Your app handles an authentication success response
+
+After the user successfully verifies their identity, Identity Engine sends an interaction code in a query parameter to `${signInRedirectURI}`. For example, `http://localhost:44314/interactioncode/callback?interaction_code=2JFmObNY8snovJP6_UK5gI_l7RQ-....`
```
[the remark this answers is not in the document]
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md:27]: It occurs earlier than this, but yes, I have done so.

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/index.md [hunk 8]
```diff
@@ -2,14 +2,16 @@
 title: Sign in with email only
 ---
 
+<div class="oie-embedded-sdk">
+
 <ApiLifecycle access="ie" /><br>
 
 Enable users to sign in with email only in an application using the embedded Okta Sign-In Widget.
```
[the remark this answers is not in the document]
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/index.md:9]: @chrisdmills-okta How about "Enable an email-only sign-in flow in your application using the embedded Okta Sign-In Widget."

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md (hunk 1 above)
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md:29]: Fixed. @robertnicolo-okta  Are you happy for me to make this word change in all our pwd-optional guides? We have the same text everywhere.

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md (hunk 5 above)
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md:1]: Changed to "The following code defines an endpoint that receives the `OTP` and `state` values from the magic link. It then attempts to retrieve the current Sign-In Widget configuration details from session state. If successful, the `OTP` and `state` values are added to the configuration details and passed to the Widget. If unsuccessful, the code assumes the callback has not come from the same browser and prompts the user accordingly."

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/nodeexpress/bestpractices.md (hunk 2 above)
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/nodeexpress/bestpractices.md:1]: fixed

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/react/bestpractices.md (hunk 3 above)
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/react/bestpractices.md:1]: fixed

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/bestpractices.md (hunk 4 above)
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/bestpractices.md:1]: fixed

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md]: Yup, that works just fine.

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md (hunk 1 above)
Comment (robertnicolo-okta) [reply packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md]: Yes, @danielmaharry-okta

**Review by robertnicolo-okta — APPROVED** [review]
lgtm

**Pull request merged**